tests/Application/Commands/RedistibuteMatchFundsTest.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MatchBot\Tests\Application\Commands;
+
+use MatchBot\Application\Commands\RedistributeMatchFunds;
+use MatchBot\Application\HttpModels\DonationCreate;
+use MatchBot\Domain\Campaign;
+use MatchBot\Domain\CampaignFunding;
+use MatchBot\Domain\CampaignFundingRepository;
+use MatchBot\Domain\ChampionFund;
+use MatchBot\Domain\Donation;
+use MatchBot\Domain\DonationRepository;
+use MatchBot\Domain\FundingWithdrawal;
+use MatchBot\Domain\Pledge;
+use MatchBot\Tests\TestCase;
+use Prophecy\Argument;
+use Prophecy\Prophecy\ObjectProphecy;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\Console\Tester\CommandTester;
+use Symfony\Component\Lock\LockFactory;
+
+class RedistibuteMatchFundsTest extends TestCase
+{
+    public function testNoEligibleDonations(): void
+    {
+        $now = new \DateTimeImmutable('now');
+
+        $donationRepoProphecy = $this->prophesize(DonationRepository::class);
+        $donationRepoProphecy->findWithMatchingWhichCouldBeReplacedWithHigherPriorityAllocation(
+            $now,
+            $now->sub(new \DateInterval('P6W')),
```

Comment:
IMHO one of the main reasons to DI the time is to allow us to be specific about what scenario we're testing with regards to dates & times, and avoid repeating code from the SUT in the test. Maybe something like this: https://github.com/thebiggive/matchbot/compare/MAT-342-redistribute-matching...MAT-342-redistribute-matching-use-specific-datetime-in-test?expand=1 ?

I don't know if it's possible to do the same with the integration test - it might be, or that might require wider refactoring.

Reply:
Thanks, I've merged that branch into this.

Integration tests just use `now` already so I think that short of injecting a clock they're probably as good as they can be. I think there is also an advantage in using the 'real' clock in that case so I'm not sure I would significantly change what is happening even if a clock were being injected.